### ISSUE TITLE
Uploading to S3 with BucketKeyEnabled throws an error

### ISSUE THREAD

**Issue by humancopy** [aws/aws-sdk-php#2251]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/aws-sdk-php/v3/api/index.html)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-php-sdk) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-php/issues) and didn't find any solution
  
**Describe the bug**
Trying to upload to `S3` with `"BucketKeyEnabled" => true`  throws an error.

**Version of AWS SDK for PHP?**
v3.183.7

**Version of PHP (`php -v`)?**
v8.0.2

**To Reproduce (observed behavior)**
```php
$uploader = new Aws\S3\ObjectUploader(
    $client,
    $bucket,
    $key,
    $source,
    'private',
    [
        'params' => [
            'ServerSideEncryption' => 'aws:kms',
            'BucketKeyEnabled'     => true,
        ]
    ]
);
```

**Expected behavior**
The file would get uploaded and the `BucketKeyEnabled` should be enabled on it.

**Screenshots**
Throws an error:
```
An uncaught Exception was encountered
Type: Aws\S3\Exception\S3Exception

Message: Error executing "PutObject" on "..."; AWS HTTP error: Client error: `PUT ...` resulted in a `400 Bad Request` response: InvalidArgumentValue should be either 'true' or 'fal (truncated...) InvalidArgument (client): Value should be either 'true' or 'false' - InvalidArgumentValue should be either 'true' or 'false'x-amz-server-side-encryption-bucket-key-enabled4K7BDPJMK4Z88YP9JBIz94Iy3WjUsNGJpE53SPl8JScPVJW6VzOQYy04oMymLEvWYxpnFT2zdgg+s2nTmujPb4OeWtY=

Filename: /var/www/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php

Line Number: 195

Backtrace:

File: /var/www/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php
Line: 97
Function: parseError

File: /var/www/vendor/guzzlehttp/promises/src/Promise.php
Line: 204
Function: Aws\{closure}

File: /var/www/vendor/guzzlehttp/promises/src/Promise.php
Line: 169
Function: callHandler

File: /var/www/vendor/guzzlehttp/promises/src/RejectedPromise.php
Line: 42
Function: GuzzleHttp\Promise\{closure}

File: /var/www/vendor/guzzlehttp/promises/src/TaskQueue.php
Line: 48
Function: GuzzleHttp\Promise\{closure}

File: /var/www/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php
Line: 158
Function: run

File: /var/www/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php
Line: 183
Function: tick

File: /var/www/vendor/guzzlehttp/promises/src/Promise.php
Line: 248
Function: execute

File: /var/www/vendor/guzzlehttp/promises/src/Promise.php
Line: 224
Function: invokeWaitFn

File: /var/www/vendor/guzzlehttp/promises/src/Promise.php
Line: 269
Function: waitIfPending

File: /var/www/vendor/guzzlehttp/promises/src/Promise.php
Line: 226
Function: invokeWaitList

File: /var/www/vendor/guzzlehttp/promises/src/Promise.php
Line: 269
Function: waitIfPending

File: /var/www/vendor/guzzlehttp/promises/src/Promise.php
Line: 226
Function: invokeWaitList

File: /var/www/vendor/guzzlehttp/promises/src/Promise.php
Line: 62
Function: waitIfPending

File: /var/www/vendor/aws/aws-sdk-php/src/S3/ObjectUploader.php
Line: 91
Function: wait
```

**Comment by SamRemis** [comment]
Hi @humancopy, 
Thank you for bringing this to our attention and the PR!  Since sending a '1' or a '0' is an established behavior for our SDK, I have contacted the S3 encryption team to ask if they can change their code to accept these values.  They told me they will discuss it today.

**Comment by stobrien89** [comment]
Fixed by #2252
